src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotHardCodeEncryptionKeyTests.cs

```diff
@@ -292,6 +292,26 @@ public void CreateEncryptor(byte[] rgbKey)
             GetCSharpResultAt(16, 9, 9, 22, "byte[] SymmetricAlgorithm.Key", "void TestClass.CreateEncryptor(byte[] rgbKey)", "byte[] Convert.FromBase64String(string s)", "void TestClass.TestMethod()"));
         }
 
+        [Fact]
+        public void Test_HardcodedIn2DByteArray_CreateEncryptor_Diagnostic()
```

Comment:
>Test_HardcodedIn2DByteArray_CreateEncryptor_Diagnostic [](start = 20, length = 54)

It'd be good to add a test for a jagged array (like byte[][]) also.

Is there a test case that covers the assert from #2708?